### PR TITLE
[Improve] yarn submission mode: Check whether the same task is submitted repeatedly

### PR DESCRIPTION
**Pull request by hackallan**
What changes were proposed in this pull request
 in  ApplicationActionServiceImpl  class  the public void start(Application appParam, boolean auto) method adds the yarn mode submission task check

Brief change log
Verifying this change

This change added tests and can be verified as follows:
If a task is already running or queued in the yarn queue, it will not be submitted
image
image

Does this pull request potentially affect one of the following parts
Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/ApplicationActionServiceImpl.java [hunk 1]
```diff
@@ -381,6 +383,15 @@ public void start(Application appParam, boolean auto) throws Exception {
     ApiAlertException.throwIfTrue(
         !application.isCanBeStart(), "[StreamPark] The application cannot be started repeatedly.");
 
+    if (FlinkExecutionMode.YARN_APPLICATION == application.getFlinkExecutionMode()
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/ApplicationActionServiceImpl.java]: We can improve this code：

if (FlinkExecutionMode.isYarnMode(application.getFlinkExecutionMode())) {
...
}

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/ApplicationActionServiceImpl.java [hunk 2]
```diff
@@ -565,6 +576,28 @@ public void start(Application appParam, boolean auto) throws Exception {
             });
   }
 
+  /**
+   * Check whether a job with the same name is running in the yarn queue
+   *
+   * @param jobName
+   * @return
+   */
+  private boolean checkAppRepeatInYarn(String jobName) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/application/impl/ApplicationActionServiceImpl.java:589]: We also need to check the ApplicationType: 
```
private boolean checkAppRepeatInYarn(String jobName) {
    try {
      YarnClient yarnClient = HadoopUtils.yarnClient();
      Set<String> types =
          Sets.newHashSet(
              ApplicationType.STREAMPARK_FLINK.getName(), ApplicationType.APACHE_FLINK.getName());
      EnumSet<YarnApplicationState> states =
          EnumSet.of(YarnApplicationState.RUNNING, YarnApplicationState.ACCEPTED);
      List<ApplicationReport> reportList = yarnClient.getApplications(types, states);
      for (ApplicationReport report : reportList) {
        if (report.getName().equals(jobName)) {
          return true;
        }
      }
      return false;
    } catch (Exception e) {
      throw new RuntimeException("The yarn api is abnormal. Ensure that yarn is running properly.");
    }
  }

```

**Review by wolfboys — APPROVED** [review]
LGTM

**Comment by caicancai** [comment]
@hackallan Thank you for your contribution, great work

**Pull request merged**